scripts/west_commands/runners/jlink.py

```diff
@@ -121,6 +122,7 @@ def print_gdbserver_message(self):
         self.logger.info('J-Link GDB server running on port {}'.
                          format(self.gdb_port))
 
+    @lru_cache(maxsize=1)
```

Comment:
It is probably more optimal to cache using a variable instead, but compared to the overhead of executing a process I felt this wasn't worth it.

Reply:
is it worth doing that here? do we call this funcion frequently? version could also be cached the first time it is read or on init

Reply:
As it is, it spawns a new process and reads its output, so it's very expensive. We need to cache its result, which is immutable, one way or another, and this is the smallest way to do it in terms of lines of code.

Reply:
>version could also be cached the first time it is read
That is exactly what this line is doing